From 7bf8e76d0a903e524f65004beaec0b274a377323 Mon Sep 17 00:00:00 2001
From: Tim Yao <tim.yao@amlogic.com>
Date: Mon, 19 Jan 2015 12:10:58 -0800
Subject: [PATCH 5373/5965] PD #102102. H.264 4K video decoder: Fix MBAFF
 encoded 4k video decoding. Add max reference number calculation for level 5.2

Change-Id: I5ce485c9c577e0379b88d354701aa56c15f25e1b
---
 .../amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux.h | 4 ++--
 .../amports/m6tvd/ucode/h264_4k2k/h264slice_linux_single.h  | 4 ++--
 drivers/amlogic/amports/m8/ucode/h264_4k2k/h264c_linux.h    | 2 +-
 drivers/amlogic/amports/m8/ucode/h264_4k2k/h264mmc_linux.h  | 4 ++--
 .../amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux.h    | 6 +++---
 .../amports/m8/ucode/h264_4k2k/h264slice_linux_single.h     | 4 ++--
 drivers/amlogic/amports/vh264_4k2k.c                        | 4 ++--
 7 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux.h b/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux.h
index 5d3d9db70f5d..7a05ae29d82a 100755
--- a/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux.h
+++ b/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux.h
@@ -159,8 +159,8 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x6800088,0x7c022c8,0xc7ff8c0,0x7c02308,0x681000a,0x20092aa,0x649ce0a,0x2009289,
 	0x54012a5,0x200a94a,0x680008b,0x7c022cb,0x68000cb,0x6435008,0x580f208,0xbef8048,
 	0x6803f08,0x6035109,0x6035208,0x6a08008,0x6035008,0xac0328f,0xc7ff3c0,0x7c0230b,
-	0x6808008,0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x480f34d,0x7c0230d,
-	0x980c38c,0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x480f34d,
+	0x6808008,0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x481f34d,0x7c0230d,
+	0x980c38c,0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x481f34d,
 	0x7c022cd,0x643500d,0x580f34d,0xbef804d,0x0000000,0x643530d,0x900c34d,0xa6f800d,
 	0x0000000,0x603510b,0x6803f0b,0x603520b,0xc7fed80,0x603500c,0x780234c,0xa60c00c,
 	0x440120c,0x6800088,0xaa14008,0x6a01009,0xaa0c048,0x6a22009,0x6a40009,0x6c00109,
diff --git a/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux_single.h b/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux_single.h
index 2b39889722cb..170cb3dd18a9 100755
--- a/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux_single.h
+++ b/drivers/amlogic/amports/m6tvd/ucode/h264_4k2k/h264slice_linux_single.h
@@ -150,8 +150,8 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x7c022c8,0xc7ff8c0,0x7c02308,0x681000a,0x20092aa,0x649ce0a,0x2009289,0x54012a5,
 	0x200a94a,0x680008b,0x7c022cb,0x68000cb,0x6435008,0x580f208,0xbef8048,0x6803f08,
 	0x6035109,0x6035208,0x6a08008,0x6035008,0xac0328f,0xc7ff3c0,0x7c0230b,0x6808008,
-	0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x480f34d,0x7c0230d,0x980c38c,
-	0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x480f34d,0x7c022cd,
+	0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x481f34d,0x7c0230d,0x980c38c,
+	0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x481f34d,0x7c022cd,
 	0x643500d,0x580f34d,0xbef804d,0x0000000,0x643530d,0x900c34d,0xa6f800d,0x0000000,
 	0x603510b,0x6803f0b,0x603520b,0xc7fed80,0x603500c,0x780234c,0xa60c00c,0x440120c,
 	0x6800088,0xaa14008,0x6a01009,0xaa0c048,0x6a22009,0x6a40009,0x6c00109,0x60e0509,
diff --git a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264c_linux.h b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264c_linux.h
index 2026fc2f7f45..e631122ca327 100755
--- a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264c_linux.h
+++ b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264c_linux.h
@@ -39,7 +39,7 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x0000000,0x0800000,0x649c700,0xa60c000,0xcc00000,0x0400000,0xc000240,0x0000000,
 	0x67f2500,0x93e1000,0xaa0c000,0xc000480,0x0000000,0xcc00000,0x0400000,0x809c700,
 	0x9104040,0xaa0c001,0x6804041,0x6806041,0x48ff000,0x4401000,0x5403000,0x2000040,
-	0x7000001,0x9c01281,0x9c012c1,0xcc00000,0x7400001,0x69a0001,0x6c00781,0x6804002,
+	0x7000001,0x9c01281,0x9c012c1,0xcc00000,0x7400001,0x6b00001,0x6c00ec1,0x6804002,
 	0xc0003c0,0x6a04000,0x78000c1,0x63f2b01,0x7800101,0x63f2c01,0x7800281,0x63f2801,
 	0x7800241,0x63f2701,0x67f2501,0x4001041,0x63f2501,0xcc00000,0x0000000,0x6035202,
 	0x649ce02,0x2002042,0x6035102,0x6035000,0x6435001,0x580f041,0xbef8041,0x0000000,
diff --git a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264mmc_linux.h b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264mmc_linux.h
index 3d6ff9944adc..590fc76b6fc3 100755
--- a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264mmc_linux.h
+++ b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264mmc_linux.h
@@ -200,10 +200,10 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x901e30b,0xac0428c,0xb8fd28c,0x93c230b,0xbaec08c,0xcc00000,0x0000000,0xc05b700,
 	0x0000000,0x67f2a0b,0x901e30b,0xac0428c,0xb8fb28c,0x93c230b,0xbae408c,0xcc00000,
 	0x0000000,0x67f2a0b,0xa21804b,0x901e30b,0xb8fd28c,0xac0328c,0x93c230b,0xa6e80cc,
-	0xcc00000,0x0000000,0x69a0009,0x6c00789,0x680400a,0xc000bc0,0x6a0400b,0x78000c9,
+	0xcc00000,0x0000000,0x6b00009,0x6c00ec9,0x680400a,0xc000bc0,0x6a0400b,0x78000c9,
 	0x63f2b09,0x7800109,0x63f2c09,0x7800289,0x63f2809,0x7800249,0x63f2709,0x7801c49,
 	0x63f2309,0x7801bc9,0x7801c0a,0x961024a,0x63f2209,0x7801cc9,0x7801d0a,0x961024a,
-	0x63f2109,0xcc00000,0x0000000,0x69a0009,0x6c00789,0x680400a,0xc000580,0x6b0400b,
+	0x63f2109,0xcc00000,0x0000000,0x6b00009,0x6c00ec9,0x680400a,0xc000580,0x6b0400b,
 	0x67f2b09,0x7c000c9,0x67f2c09,0x7c00109,0x67f2809,0x7c00289,0x67f2709,0x7c00249,
 	0x67f2309,0x7c01c49,0x67f2209,0x7c01bc9,0x9210249,0x7c01c09,0x67f2109,0x7c01cc9,
 	0x9210249,0x7c01d09,0xcc00000,0x0000000,0x603520a,0x649ce0a,0x200a24a,0x603510a,
diff --git a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux.h b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux.h
index 5d3d9db70f5d..7972388c1f52 100755
--- a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux.h
+++ b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux.h
@@ -20,7 +20,7 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x4001208,0x7c02608,0x6800048,0x60e0c08,0x7c027c8,0x7c02648,0x6800008,0x7c02808,
 	0x68003c8,0x7c02688,0x6805c08,0x60c2108,0x80c2170,0x6800088,0x7c02748,0xc016b00,
 	0x0000000,0xcc00000,0x0000000,0x780280b,0xa6fc00b,0x0000000,0x6a0002b,0x6c007ab,
-	0x690002a,0x649cd08,0xbe1a008,0x6c007aa,0x6a0002b,0x6c00eeb,0x680002a,0x6c00eea,
+	0x690002a,0x649cd08,0xbe1a008,0x6c007aa,0x690002b,0x6c00eeb,0x680002a,0x6c00eea,
 	0xc016b40,0x0000000,0xc017040,0x0000000,0x6900009,0x649cd08,0xbe12008,0x6c000c9,
 	0x6800009,0x6c00809,0x78016c8,0x5c0c289,0x6819da4,0xaa18088,0x6800009,0x6816724,
 	0xaa0c148,0x6800049,0x6800089,0x7c02589,0x7801788,0x94c2248,0x60c0409,0x7c026ca,
@@ -159,8 +159,8 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x6800088,0x7c022c8,0xc7ff8c0,0x7c02308,0x681000a,0x20092aa,0x649ce0a,0x2009289,
 	0x54012a5,0x200a94a,0x680008b,0x7c022cb,0x68000cb,0x6435008,0x580f208,0xbef8048,
 	0x6803f08,0x6035109,0x6035208,0x6a08008,0x6035008,0xac0328f,0xc7ff3c0,0x7c0230b,
-	0x6808008,0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x480f34d,0x7c0230d,
-	0x980c38c,0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x480f34d,
+	0x6808008,0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x481f34d,0x7c0230d,
+	0x980c38c,0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x481f34d,
 	0x7c022cd,0x643500d,0x580f34d,0xbef804d,0x0000000,0x643530d,0x900c34d,0xa6f800d,
 	0x0000000,0x603510b,0x6803f0b,0x603520b,0xc7fed80,0x603500c,0x780234c,0xa60c00c,
 	0x440120c,0x6800088,0xaa14008,0x6a01009,0xaa0c048,0x6a22009,0x6a40009,0x6c00109,
diff --git a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux_single.h b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux_single.h
index 2b39889722cb..170cb3dd18a9 100755
--- a/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux_single.h
+++ b/drivers/amlogic/amports/m8/ucode/h264_4k2k/h264slice_linux_single.h
@@ -150,8 +150,8 @@ const u32 MicroCode[] __attribute__ ((aligned (8))) = {
 	0x7c022c8,0xc7ff8c0,0x7c02308,0x681000a,0x20092aa,0x649ce0a,0x2009289,0x54012a5,
 	0x200a94a,0x680008b,0x7c022cb,0x68000cb,0x6435008,0x580f208,0xbef8048,0x6803f08,
 	0x6035109,0x6035208,0x6a08008,0x6035008,0xac0328f,0xc7ff3c0,0x7c0230b,0x6808008,
-	0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x480f34d,0x7c0230d,0x980c38c,
-	0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x480f34d,0x7c022cd,
+	0x2009209,0x200a94a,0xc7ffc80,0x40012cb,0x400134d,0x481f34d,0x7c0230d,0x980c38c,
+	0x78022cd,0x54092cd,0x200ba8b,0x649ce0a,0x200b28b,0x400134d,0x481f34d,0x7c022cd,
 	0x643500d,0x580f34d,0xbef804d,0x0000000,0x643530d,0x900c34d,0xa6f800d,0x0000000,
 	0x603510b,0x6803f0b,0x603520b,0xc7fed80,0x603500c,0x780234c,0xa60c00c,0x440120c,
 	0x6800088,0xaa14008,0x6a01009,0xaa0c048,0x6a22009,0x6a40009,0x6c00109,0x60e0509,
diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 79a7a20dd11a..763246a4b33c 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -552,10 +552,10 @@ static int get_max_dec_frame_buf_size(int level_idc, int max_reference_frame_num
             size = 42393600;
         break;
         case 51:
+        case 52:
+        default:
             size = 70778880;
         break;
-        default:
-            break;
     }
 
     size /= pic_size;
-- 
2.19.0

